readalongs/align.py

```diff
@@ -956,54 +968,61 @@ def get_ancestor_sent_el(word_el: etree.ElementTree) -> Union[None, etree.Elemen
     return word_el
 
 
-def get_words_and_sentences(results) -> Tuple[List[dict], List[List[dict]]]:
-    """Parse xml into word and sentence 'tier' data
+def get_textual_words_and_sentences(
+    words: List[dict], tokenized_xml: etree.ElementTree
+) -> Tuple[List[dict], List[List[dict]]]:
+    """Parse xml into word and sentence 'tier' data with full textual words
 
     Args:
-        results([TODO type]): the results object returned by align_audio
+        words: list of words with "id", "start" and "end"
+        tokenized_xml: tokenized or g2p'd parsed XML object
 
     Returns:
-        list of words, list of sentences
+        list of words, list of sentences (as a list of lists of words)
+        The returned words are dicts containing:
```

Comment:
Orthogonal to this PR, but it would be more robust to use a NamedTuple or data class instead of a dict here.

Reply:
God yes! I really don't like our dicts used like this all over the code. But I've never had the courage to change them, because they're really all over!

Reply:
this seems like a better idea (it is also what pocketsphinx and soundswallower do internally, except that of course their "text" is our "id")

Reply:
That, if and when we do it, will definitely be in a separate PR.

Reply:
But I like the idea enough that I think I'll do it.